### ISSUE TITLE
🔍 [DISCOVERY] - Basics of JSON

### ISSUE THREAD

**Issue by mihir-bombay-studio** [dear-digital/linter#54]
### Is there an existing Discovery issue on this topic?

- [X] I have searched the existing issues

### Objective

Use chatGPT or similar LLMs available to ask the following question.
"Explain what JSON is and why it's important."
and follow up questions like:
"What are common use-cases for JSON?"

### Expected Outcome

List down only the key insights gained from multiple rounds of questioning.
List all the questions/prompts used below in the comment section. **You'll directly get points rewarded for this task for adding correct answer as comment**

### Example on how to write outcome

> #### Initial Prompt:
> - "Describe what linting is and why it's important in software development."
> 
> #### Recursive Prompts:
> 1. "How do linters work under the hood?"
> 2. "What types of problems can linters catch?"
> 3. "Explain the impact of linting on code quality."
> 4. "How can linting be integrated into a continuous integration (CI) system?"
> .
> .
> .
> 
> #### Outcome:
> 
> ##### Key Insights gained:
> - Linting is the process of checking source code for programmatic and stylistic errors.
> - Linters use static analysis to identify issues without running the code.
> - They can catch syntax errors, potential bugs, or even style guide violations.
> - Linting contributes to better code readability and maintainability.
> - It can be automated and integrated into CI pipelines for consistent code quality.
> .
> .
> .
> 

### Have you provided comprehensive details for this discovery task?

- [X] Yes, I have provided comprehensive details

**Comment by abhishekjani08** [comment]
Initial Prompt:

- "Describe what linting is and why it's important in software development."

Recursive Prompts:

1. What is JSON?
2. Why is JSON important?
3. What is the structure of JSON?
4. Are there any arrays in JSON?
5. How can JSON be used in web development?
6. What is the difference between JSON and XML?
7. Can JSON handle nested structures?
8. Are there any limitations to JSON?
9. How do you validate JSON data?
10. What are the practical Uses of JSON?


Outcome:

## What is JSON
JSON (JavaScript Object Notation) is a lightweight and human-readable data interchange format. It's commonly used for data storage, configuration files, and data exchange between a server and a client, making it an essential part of modern web development.

## Why JSON?

JSON is important for several reasons:

- **Simplicity**: JSON uses a straightforward and easy-to-understand syntax, making it accessible to both humans and machines.

- **Readability**: JSON data is organized in a way that is easy for humans to read and write. This makes it a preferred choice for configuration files and data storage.

- **Data Interchange**: JSON serves as a common language for data exchange between different programming languages. It's supported by a wide range of programming languages, making it an excellent choice for web APIs.

- **Lightweight**: JSON is a lightweight format, meaning it doesn't add significant overhead to data transmission or storage.

- **Extensibility**: JSON can be extended to accommodate custom data types and structures, making it versatile for various use cases.

## JSON Structure

JSON data is organized into key-value pairs, similar to how objects work in many programming languages. Here's a basic example of JSON data:

```json
{
  "name": "John Doe",
  "age": 30,
  "city": "New York"
}
```
## Arrays in JSON
JSON also supports arrays, which are ordered collections of values. Here's an example of a JSON array:

```json
{
  "fruits": ["apple", "banana", "cherry"]
}
```

## Can JSON Handle Nested Structures?

Yes, JSON (JavaScript Object Notation) is capable of handling nested structures. This feature allows you to represent complex and hierarchical data relationships in a structured format. Let's explore an example to illustrate this concept:

```json
{
  "person": {
    "name": "John Doe",
    "age": 30,
    "address": {
      "street": "123 Main St",
      "city": "New York",
      "zipcode": "10001"
    },
    "hobbies": ["reading", "traveling"]
  }
}
```

## JSON vs. XML

JSON (JavaScript Object Notation) and XML (eXtensible Markup Language) are both data interchange formats used in various contexts, including web development. While they serve similar purposes, they have distinct differences:


| **Feature**    | **JSON**                                                  | **XML**                                                    |
| -------------- | --------------------------------------------------------- | ---------------------------------------------------------- |
| **Syntax**     | JSON uses a simple and compact syntax with key-value pairs, making it easy for humans to read and write. | XML uses a tag-based syntax with open and close tags, which can make it more verbose compared to JSON. |
| **Example**    | ``` { "name": "John Doe", "age": 30, "city": "New York" } ``` | ``` <person> <name>John Doe</name> <age>30</age> <city>New York</city> </person> ``` |
| **Readability**| JSON is highly readable for both humans and machines, which makes it a preferred choice for data interchange and configuration files. | XML can be less human-readable due to its verbosity, but it provides clear structure and validation capabilities. |
| **Usage**      | JSON is commonly used in web development for APIs, configuration, and data storage. | XML is used in a variety of contexts, including data exchange, configuration files, and document markup (e.g., HTML, SVG). |

## Limitations of JSON

Yes, JSON has some limitations:

- **Lack of Complex Data Types**: JSON supports basic data types such as strings, numbers, booleans, arrays, and objects. However, it lacks more complex data types like dates and binary data.

- **No Comments**: JSON does not support comments within the data itself, which can be inconvenient for adding explanatory notes.

- **No Circular References**: JSON cannot represent circular references in data structures, which can be a limitation in certain scenarios.

## Practical Uses of JSON

JSON's versatility makes it invaluable in various domains:

- **Web Development**: JSON is widely used for data exchange between web servers and clients, making it essential for web APIs.

- **Configuration Files**: JSON is employed to store application configurations, user preferences, and settings in a structured format.

- **Data Storage**: JSON is ideal for storing structured data in databases and file systems, enabling easy access and manipulation.

- **Interoperability**: JSON's simplicity and compatibility with multiple programming languages make it perfect for data interchange and integration between different systems.


_In conclusion, JSON plays a fundamental role in the digital landscape, simplifying data handling and fostering interoperability across the software development ecosystem._

**Comment by mihir-bombay-studio** [comment]
Thanks @abhishekjani08
